Flaky test: test_run_and_cancel_legacy_ensemble

Got this while running Github Actions. Unsure why. Python 3.9 on Ubuntu.
```
=================================== FAILURES ===================================
_____________________ test_run_and_cancel_legacy_ensemble ______________________

tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_run_and_cancel_legacy_ens0')
unused_tcp_port = 57351
make_ensemble_builder = <function make_ensemble_builder.<locals>._make_ensemble_builder at 0x7fcc2917adc0>

    @pytest.mark.timeout(60)
    def test_run_and_cancel_legacy_ensemble(tmpdir, unused_tcp_port, make_ensemble_builder):
        num_reals = 10
        with tmpdir.as_cwd():
            ensemble = make_ensemble_builder(tmpdir, num_reals, 2, job_sleep=5).build()
            config = EvaluatorServerConfig(unused_tcp_port)
    
            evaluator = EnsembleEvaluator(ensemble, config, 0, ee_id="1")
    
            with evaluator.run() as mon:
                cancel = True
                for _ in mon.track():
                    if cancel:
                        mon.signal_cancel()
                        cancel = False
    
>           assert evaluator._ensemble.get_status() == state.ENSEMBLE_STATE_CANCELLED
E           AssertionError: assert 'Stopped' == 'Cancelled'
E             - Cancelled
E             + Stopped

/home/runner/work/ert/ert/tests/ensemble_evaluator/test_ensemble_legacy.py:50: AssertionError
------------------------------ Captured log call -------------------------------
WARNING  ert_shared.ensemble_evaluator.entity.ensemble_base:ensemble_base.py:53 Illegal state transition from Cancelled to Stopped
```

--- a/res/job_queue/queue.py
+++ b/res/job_queue/queue.py
@@ -451,7 +451,7 @@ class JobQueue(BaseCClass):
             await JobQueue._publish_changes(ee_id, self.snapshot(), websocket)
 
             try:
-                while self.is_active() and not self.stopped:
+                while True:
                     self.launch_jobs(pool_sema)
 
                     await asyncio.sleep(1)
@@ -463,6 +463,8 @@ class JobQueue(BaseCClass):
                     await JobQueue._publish_changes(
                         ee_id, self._changes_after_transition(), websocket
                     )
+                    if not self.is_active() or self.stopped:
+                        break
             except asyncio.CancelledError:
                 if self.stopped:
                     logger.debug(

